docs/deployment/aks.md

```diff
@@ -0,0 +1,592 @@
+installing the AKS cluster, we'll install stable/nginx-ingress helm chart as
+you can see below.
+
+```
+az group create --name garden-1 --location eastus
+az aks create --resource-group garden-1 --name garden-1 --node-count 3 --generate-ssh-keys -s Standard_DS4_v2
+az aks get-credentials --resource-group garden-1 --name garden-1
+```
+
+### Install default ClusterRoles
+
+On Azure, RBAC support is planned for GA and not enabled yet, so
+ClusterRole definitions are not in place and this causes Gardener helm
+chart to fail. Below instead of removing RBAC support from Gardener
+installation we are faking the RBAC support by manually creating ClusterRoles
+on our AKS cluster.
+
+**Note**: We are using v1.9.6 in our AKS cluster, mind the version in
+the url below and please use the rules for the right version.
+
+```
+# manually create ClusterRoles in AKS, required until RBAC support is added to AKS
+kubectl apply -f https://raw.githubusercontent.com/kubernetes/kubernetes/v1.9.6/plugin/pkg/auth/authorizer/rbac/bootstrappolicy/testdata/cluster-roles.yaml --validate=false
+```
+
+### Install helm/tiller to AKS
+
+Helm is also need to be run with a ClusterRole (cluster-admin) with
+enough permissions even if RBAC is not enabled, this is an enforcement
+needs to be addressed on Helm side not on AKS/Kubernetes. So lets
+create a tiller ServiceAccount with cluster-admin ClusterRole and
+deploy Helm with the ServiceAccount
+([`example/aks/tiller-rbac-config.yaml`](../../example/aks/tiller-rbac-config.yaml)).
+
+```
+# Reference: https://github.com/kubernetes/helm/blob/master/docs/rbac.md
+kubectl apply -f example/aks/tiller-rbac-config.yaml
+helm init --service-account tiller
+```
+
+### Deploy stable/nginx-ingress chart to AKS
+
+At the moment all `Ingress` resources created by the Gardener are
+using the nginx annotations. This is the simplest solution at the
+moment, because it works across cloud providers. But in the future
+it's planned to make it more customizable.
+
+[AKS Ingress](https://docs.microsoft.com/en-us/azure/aks/ingress)
+documentation tells about installing Ingress controller with RBAC
+disabled, you can also follow that one but since we already deployed
+the default ClusterRoles above, we are now able to install the chart
+with RBAC enabled.
+
+```
+helm upgrade --install --namespace kube-system nginx-ingress stable/nginx-ingress
+```
+
+Gardener deploys multiple addons, like monitoring apps ([prometheus](https://prometheus.io/),
+[grafana](https://grafana.com/)) in shoot cluster namespaces on the seed cluster. These
+will provide some ingress addresses like p.shoot.project.garden-1.your.domain.here. But
+gardener does not manage this dns records for you. It expects you to have \*.shoot.project.garden-1.your.domain.here
+wildchar dns records per shoot cluster to be available. For that to work, we need to get
+the loadbalancer's ip address and create these records ourselves.
```

Comment:
Wow i missed this fact when going over the installation for docs, I'm quite a bit surprised here about the fact that the gardener doesn't create a wildcard record for these.

Can anyone with better gardener knowledge elaborate on the design decision for the gardener not to create the seed's assumed wildcard ingress DNS record in route53? We should add it as a comment here.

Reply:
@mvladev @swapnilgm @rfranzke do you have any comments on this one?

Reply:
Sorry for the late reply, the week was quite busy.

Currently, a Seed cluster is just a Kubeconfig for the Gardener. The cluster could have been created by any tool, Gardener only cares about having a valid Kubeconfig to talk to its API. Additional to that it requires the Seed to run a nginx-ingress-controller having a wildcard DNS record already registered that is pointing to its exposed load balancer. Due to this assumption Gardener does not need to create DNS records for the Ingresses it exposes in the Seed (because the Ingress wildcard domain is known to it (https://github.com/gardener/gardener/blob/master/example/seed-azure.yaml#L14)).

In #201 @fsniper suggests to drop these two assumptions (nginx-ingress installed + wildcard DNS record already created) and let the Gardener do it on its own. I have put my thoughts as a comment to this issue.